lms/resources/lti_launch.py

```diff
@@ -38,31 +38,14 @@ def get_or_create_course(self):
         tool_consumer_instance_guid = params["tool_consumer_instance_guid"]
         context_id = params["context_id"]
 
-        authority_provided_id = course_service.generate_authority_provided_id(
-            tool_consumer_instance_guid, context_id
-        )
-
-        legacy_course = course_service.get_or_create(authority_provided_id)
-        # Capture the state here, before any other SQLA query does an implicit flush and then
-        # removing legacy_course from `db.new`
-        is_new_legacy_course = legacy_course in self._request.db.new
-
-        course = course_service.upsert(
-            authority_provided_id,
+        return course_service.upsert(
+            course_service.generate_authority_provided_id(
+                tool_consumer_instance_guid, context_id
+            ),
             context_id,
             params["context_title"],
             self._course_extra(),
-            legacy_course.settings,
```

Comment:
This call has a lot of arguments. I think it might be easier to piece together with key word args rather than positional.

This all passes the nod test if everything is in the right place, but it's very hard to spot ordering errors.